main.bundle.js

```diff
@@ -68,35 +68,150 @@
+	    const data = { [attrName]: $(this).text() };
+
+	    if (event.keyCode === 13) {
+	      ajaxPatchRequest(data, foodId);
+	      return false;
+	    }
+	  });
+	}
+
+	function ajaxPatchRequest(data, foodId) {
+	  $.ajax({
+	    type: 'PATCH',
+	    url: `${herokuUrl()}api/v1/foods/${foodId}`,
+	    data: data,
+	    success: function () {
+	      alert("Successful");
+	    },
+	    error: function () {
+	      alert("Nope");
+	    }
+	  });
+	}
+
+	function foodFilterListenter() {
+	  $(`.food-filter`).on('keydown', function (event) {
+	    let searchValue = $('.search-value').val().toLowerCase();
+	    $.ajax({
+	      type: 'GET',
+	      url: `${herokuUrl()}api/v1/foods`,
+	      success: function (posts) {
+	        appendFood(posts, searchValue);
+	      }
+	    });
+	  });
+	}
+
+	function appendFood(foods, searchValue) {
+	  let foodObjects = foods.filter(function (food) {
+	    return food.name.toLowerCase().includes(searchValue);
+	  });
+
+	  $('.food').remove();
+	  $('.deleteButton').hide();
+
+	  fillTable(foodObjects);
+	}
+
+	function fillTable(foods) {
+	  foods.reverse().forEach(function (foodType) {
+	    event.preventDefault;
+	    let button = `<td><button type='button' class='deleteButton' id='${foodType.id}'>Delete</button></td>`;
+	    let insertName = `<tr id=${foodType.id}><td class="food" name="name" contentEditable>${foodType.name}</td>`;
+	    let insertCals = `<td class="food" name="calories" contentEditable>${foodType.calories}`;
+	    let insertRow = `${insertName}${insertCals}</td>${button}</tr>`;
+	    $(".foodsTable").append(insertRow);
+	  });
+	}
+
+	function createFoodButtonListener() {
+	  $('#createButton').on('click', function () {
+	    $('#newFoodForm').show();
+	    createFoodFormListener();
+	  });
+	}
+
+	function createFoodFormListener() {
+	  $('#createFoodForm').on('submit', function (event) {
+	    event.preventDefault();
+
+	    let food = $(event.currentTarget.name).val();
+	    let calories = $(event.currentTarget.calories).val();
+
+	    errorCheck(food, calories);
+
+	    let data = $.post(herokuUrl() + `api/v1/foods?name=${food}&calories=${calories}"`).then(function (data) {
+	      let button = `<td><button type='button' class='deleteButton' id='${data.id}'>Delete</button></td>`;
+	      let toInsert = `<tr id=${data.id}><td>${food}</td><td>${calories}</td>${button}</tr>`;
+	      $(".foodsTable").prepend(toInsert);
+	      $('#createFoodForm')[0].reset();
+	    });
+	  });
+	}
+
+	function errorCheck(food, calories) {
+	  //let words = [food, calories]
+	  //let toSay = {
+	  //food: "food name",
+	  //calories: "calorie amount"
+	  //}
+
+	  //words.forEach(function(input, index) {
+	  //let word = ["food", "calories"]
+	  //if (!`${input}`) {
+	  //$(`${word[index]}-error`).remove()
+	  //throw new Error ($(`#${word[index]}Field`).append(`<p id=``${word[index]}-error``>Please enter a ${toSay[word[index]]} </p>`))
+	  //}
+	  //})
+	  //}
+
+	  if (!food) {
```

Comment:
I think this can be refactored to be more DRY.  As you can see I attempted a version that doesn't work in the commented out text above this.